echemdb/data/cv/entry.py

```diff
@@ -112,41 +112,161 @@ def __getitem__(self, name):
+
+        TODO: Adapt along with https://github.com/echemdb/svgdigitizer/issues/106.
+
+        EXAMPLES::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.x()
+            'U'
+
+        """
+        from astropy import units as u
+        if u.Unit(self.figure_description.potential_scale.unit).is_equivalent('V'):
+            return 'U'
+
+    def y(self):
+        r"""Return the y-axis name.
+
+        TODO: Adapt along with https://github.com/echemdb/svgdigitizer/issues/106.
+
+        EXAMPLES::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.y()
+            'j'
+
+        """
+        from astropy import units as u
+
+        if u.Unit(self.figure_description.current.unit).is_equivalent('A / m2'):
+            return 'j' 
+        if u.Unit(self.figure_description.current.unit).is_equivalent('A'):
+            return 'I'
+
+    def x_unit(self, xunit=None):
+        r"""Return an astropy unit, where `xunit` is a voltage.
+
+        EXAMPLES:
+
+        Without parameters, SI units are returned::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.x_unit()
+            Unit("V")
+
+        When set to `"original"`, the original units of the published figure are returned::
+
+            >>> entry.x_unit(xunit='original')
+            Unit("V")
+
+        Units can be specified explicitly::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.x_unit(xunit='mV')
+            Unit("mV")
+
+        """
+        from astropy import units as u
+
+        if xunit is None:
+            xunit = u.V
+        if xunit == 'original':
+            xunit = self.figure_description.potential_scale.unit
+        if isinstance(xunit, str):
+            xunit = u.Unit(xunit)
+
+        return xunit
+
+    def y_unit(self, yunit=None):
+        r"""Return an astropy unit, where `yunit` is a current (I) or current density (j).
+
+        EXAMPLES:
+
+        Without parameters, SI units are returned::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.y_unit()
+            Unit("A / m2")
+
+        When set to `"original"`, the original units of the published figure are returned::
+
+            >>> entry.y_unit(yunit='original')
+            Unit("mA / cm2")
+
+        Units can be specified explicitly::
+
+            >>> entry = Entry.create_examples()[0]
+            >>> entry.y_unit(yunit='uA / cm2')
+            Unit("uA / cm2")
+
+        """
+        from astropy import units as u
+
+        if yunit is None:
+            if self.y() == 'j':
+                yunit = u.A / u.m**2
+            elif self.y() == 'I':
+                yunit = u.A
+            else:
+                raise NotImplementedError("Unexpected naming of y axis.")
+
+        if yunit == 'original':
+            yunit = self.figure_description.current.unit
+
+        if isinstance(yunit, str):
+            yunit = u.Unit(yunit)
+
+        return yunit
```

Comment:
```suggestion
```suggestion
        return u.Unit(yunit)
```

`yunit` can already be a unit. `isinstance` is usually frowned upon in Python as it does not allow duck typing.